rules/windows/command_and_control_ingress_transfer_bits.toml

```diff
@@ -115,27 +120,51 @@ references = ["https://attack.mitre.org/techniques/T1197/"]
+tags = [
+    "Domain: Endpoint",
+    "OS: Windows",
+    "Use Case: Threat Detection",
+    "Tactic: Defense Evasion",
+    "Tactic: Command and Control",
+    "Data Source: Elastic Defend",
+]
 timestamp_override = "event.ingested"
 type = "eql"
 
 query = '''
 file where host.os.type == "windows" and event.action == "rename" and
-  process.name : "svchost.exe" and file.Ext.original.name : "BIT*.tmp" and 
-  (file.extension : ("exe", "zip", "rar", "bat", "dll", "ps1", "vbs", "wsh", "js", "vbe", "pif", "scr", "cmd", "cpl") or
-   file.Ext.header_bytes : "4d5a*") and 
- 
-  /* noisy paths, for hunting purposes you can use the same query without the following exclusions */
-  not file.path : ("?:\\Program Files\\*", "?:\\Program Files (x86)\\*", "?:\\Windows\\*", "?:\\ProgramData\\*\\*") and 
- 
-  /* lot of third party SW use BITS to download executables with a long file name */
-  not length(file.name) > 30 and
-  not file.path : (
-        "?:\\Users\\*\\AppData\\Local\\Temp*\\wct*.tmp",
-        "?:\\Users\\*\\AppData\\Local\\Adobe\\ARM\\*\\RdrServicesUpdater*.exe",
-        "?:\\Users\\*\\AppData\\Local\\Adobe\\ARM\\*\\AcroServicesUpdater2_x64.exe",
-        "?:\\Users\\*\\AppData\\Local\\Docker Desktop Installer\\update-*.exe"
-  )
+
+  /* svchost.exe is the process that creates BITS jobs */
+  /* original file is a temporary file created by BITS */
+  process.name : "svchost.exe" and
+  (
+    startsWith(file.Ext.original.name, "BIT") and file.extension == "tmp"
+  ) or
+  (
+    /* alternatively the original file has a PE header with a common executable extensions */
+    file.extension : ("exe", "zip", "rar", "bat", "dll", "ps1", "vbs", "wsh", "js", "vbe", "pif", "scr", "cmd", "cpl") or
+    startsWith(file.Ext.header_bytes, "4d5a") and
+
+    /* noisy paths, for hunting purposes you can use the same query without the following exclusions */
+    not (
+      startsWith(file.path, "?:\\Program Files\\") or
+      startsWith(file.path, "?:\\Program Files (x86)\\") or
+      startsWith(file.path, "?:\\Windows\\") or
+      startsWith(file.path, "?:\\ProgramData\\")
+        ) and
+
+    /* lot of third party SW use BITS to download executables with a long file name */
+    not length(file.name) > 30 and
+    not (
+      startsWith(file.path, "?:\\Users\\") and
+          (
+            (stringContains(file.path, "AppData\\Local\\Temp") and startsWith(file.name, "wct") and file.extension == "tmp") or
+            (stringContains(file.path, "AppData\\Local\\Adobe\\ARM") and startsWith(file.name, "RdrServicesUpdater") and file.extension == "exe") or
+            (stringContains(file.path, "AppData\\Local\\Adobe\\ARM") and startsWith(file.name, "AcroServicesUpdater2_") and file.extension == "exe") or
+            (stringContains(file.path, "AppData\\Local\\Docker Desktop Installer") and startsWith(file.name, "update-") and file.extension == "exe")
```

Comment:
Tbh I don't see how doing 3 wildcard conditions separated can be more performant than just one with 3 `*`, I think we should maintain the original query.

Also using these functions makes the query more complex to understand, especially because users are probably not familiar with them

Reply:
++